diff --git a/node_modules/cosmiconfig/dist/loaders.js b/node_modules/cosmiconfig/dist/loaders.js
index b7132fb..f66c167 100644
--- a/node_modules/cosmiconfig/dist/loaders.js
+++ b/node_modules/cosmiconfig/dist/loaders.js
@@ -20,7 +20,7 @@ function loadJson(filepath        , content        )         {
 }
 
 function loadYaml(filepath        , content        )         {
-  return yaml.safeLoad(content, { filename: filepath });
+  return yaml.load(content, { filename: filepath });
 }
 
 module.exports = {
